v2/internal/crdmanagement/crds_test.go

```diff
@@ -0,0 +1,308 @@
+			Group: "testrp.azure.com",
+			Scope: apiextensions.NamespaceScoped,
+			Versions: []apiextensions.CustomResourceDefinitionVersion{
+				{
+					Name:    "v1alpha1api20181130",
+					Served:  true,
+					Storage: true,
+					Schema: &apiextensions.CustomResourceValidation{
+						OpenAPIV3Schema: &apiextensions.JSONSchemaProps{
+							Properties: map[string]apiextensions.JSONSchemaProps{
+								"apiVersion": {
+									Type:        "string",
+									Description: "APIVersion description",
+								},
+								"kind": {
+									Type:        "string",
+									Description: "Kind description",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func Test_LoadCRDs(t *testing.T) {
+	t.Parallel()
+	g := NewGomegaWithT(t)
+
+	crd := makeBasicCRD("test")
+	dir := t.TempDir()
+	logger := testcommon.NewTestLogger(t)
+
+	bytes, err := yaml.Marshal(crd)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	crdPath := filepath.Join(dir, "crd.yaml")
+	g.Expect(os.WriteFile(crdPath, bytes, 0600)).To(Succeed())
+
+	crdManager := crdmanagement.NewManager(logger, nil)
+
+	loadedCRDs, err := crdManager.LoadOperatorCRDs(dir, "azureserviceoperator-system")
+	g.Expect(err).ToNot(HaveOccurred())
+
+	g.Expect(loadedCRDs).To(HaveLen(1))
+	g.Expect(loadedCRDs[0]).To(Equal(crd))
+}
+
+func Test_FindMatchingCRDs_EqualCRDsCompareAsEqual(t *testing.T) {
+	t.Parallel()
+	g := NewGomegaWithT(t)
+
+	existingCRD := makeBasicCRD("test")
+	goalCRD := makeBasicCRD("test")
+	existing := []apiextensions.CustomResourceDefinition{existingCRD}
+	goal := []apiextensions.CustomResourceDefinition{goalCRD}
+
+	logger := testcommon.NewTestLogger(t)
+	crdManager := crdmanagement.NewManager(logger, nil)
+
+	matching := crdManager.FindMatchingCRDs(existing, goal, crdmanagement.SpecEqual)
+
+	g.Expect(matching).To(HaveLen(1))
+}
+
+func Test_FindMatchingCRDs_MissingCRD(t *testing.T) {
+	t.Parallel()
+	g := NewGomegaWithT(t)
+
+	goalCRD := makeBasicCRD("test")
+	var existing []apiextensions.CustomResourceDefinition
+	goal := []apiextensions.CustomResourceDefinition{goalCRD}
+
+	logger := testcommon.NewTestLogger(t)
+	crdManager := crdmanagement.NewManager(logger, nil)
+
+	matching := crdManager.FindMatchingCRDs(existing, goal, crdmanagement.SpecEqual)
+
+	g.Expect(matching).To(BeEmpty())
+}
+
+func Test_FindMatchingCRDs_CRDsWithDifferentConversionsCompareAsEqual(t *testing.T) {
+	t.Parallel()
+	g := NewGomegaWithT(t)
+
+	existingCRD := makeBasicCRD("test")
+	existingCRD.Spec.Conversion = &apiextensions.CustomResourceConversion{
+		Strategy: apiextensions.WebhookConverter,
+		Webhook: &apiextensions.WebhookConversion{
+			ClientConfig: &apiextensions.WebhookClientConfig{
+				Service: &apiextensions.ServiceReference{
+					Name:      "azureserviceoperator-webhook-service",
+					Namespace: "azureserviceoperator-system",
+					Path:      to.StringPtr("/convert"),
+					Port:      to.Int32Ptr(443),
+				},
+				CABundle: []byte{17, 14, 12, 21, 33, 61, 25, 99, 111},
```

Comment:
Magic numbers?

Reply:
Hah, yes -- let me leave a comment here. This is just "some random byte data to simulate a CRD bundle".

Reply:
Ah, your function idea below is even better. Will do that.